src/main/java/org/datavyu/views/VideoController.java

```diff
@@ -349,6 +352,10 @@ public Identifier openVideo(final File videoFile, final Plugin plugin){
                             Datavyu.getApplication().getMainFrame(),
                             false);
                     addStream(plugin.getTypeIcon(), streamViewer);
+
+                    // Add this video's information to videoInfos
+                    videoInfos.add(new VideoInfo(id, videoFile, plugin, streamViewer));
```

Comment:
Why do we need a new class to hold the video's information, we have all the information available through the ViewerSetting and StreamViewer. I would suggest re-using the existing code rather than creating a Data structure.

Reply:
This isn't used anywhere on the Java side but made it easier to create Ruby API functions for this feature. I'll see if I can move all of this to the API itself.